base/algorithms.jl

```diff
@@ -0,0 +1,119 @@
+module Algorithms
+
+export replace!, replace
+
+import Base: replace
+
+
+"""
+    replace!(pred, [f::Function], A, [new], [count])
+
+Replace all occurrences `x` in collection `A` for which `pred(x)` is true
+by `new` or `f(x)` (exactly one among `f` and `new` must be specified).
+If `count` is specified, then replace at most `count` occurrences.
+This is the in-place version of [`replace`](@ref).
+
+# Examples
+```jldoctest
+julia> a = [1, 2, 3, 1];
+
+julia> replace!(isodd, a, 0, 2); a
```

Comment:
The inconsistency with `replace` doesn't sound a big issue to me, but I would be more concerned about consistency with other in-place functions.